tests/test_postgresql_provider.py

```diff
@@ -204,3 +208,35 @@ def test_get_not_existing_item_raise_exception(config):
     p = PostgreSQLProvider(config)
     with pytest.raises(ProviderItemNotFoundError):
         p.get(-1)
+
+@pytest.mark.parametrize('cql, expected_ids', [
+  ("osm_id BETWEEN 80800000 AND 80900000",
+   [80827787, 80827793, 80835468, 80835470, 80835472, 80835474,
+    80835475, 80835478, 80835483, 80835486]),
+  ("osm_id BETWEEN 80800000 AND 80900000 AND waterway = 'stream'",
+   [80835470]),
+  ("osm_id BETWEEN 80800000 AND 80900000 AND waterway ILIKE 'sTrEam'",
+   [80835470]),
+  ("osm_id BETWEEN 80800000 AND 80900000 AND waterway LIKE 's%'",
+   [80835470]),
+  ("osm_id BETWEEN 80800000 AND 80900000 AND name IN ('Muhira', 'Mpanda')",
+   [80835468, 80835472, 80835475, 80835478]),
+  ("osm_id BETWEEN 80800000 AND 80900000 AND name IS NULL",
+   [80835474, 80835483]),
+  ("osm_id BETWEEN 80800000 AND 80900000 AND BBOX(foo_geom, 29, -2.8, 29.2, -2.9)",
+   [80827793, 80835470, 80835472, 80835483, 80835489]),
+  ("osm_id BETWEEN 80800000 AND 80900000 AND "
+   "CROSSES(foo_geom,  LINESTRING(29.091 -2.731, 29.253 -2.845))",
+   [80835470, 80835472, 80835489])
+])
+def test_query_cql(config, cql, expected_ids):
```

Comment:
Can you also test if the default filter mechanism based on properties is used then the backend query is updated accordingly?

Reply:
See comment: https://github.com/geopython/pygeoapi/pull/964#issuecomment-1235265715

We will update this in the current round of refactoring.

Reply:
We have added a test case that uses the CQL filter combined with more traditional BBOX and properties filters. https://github.com/BritishGeologicalSurvey/pygeoapi/blob/8fead7f91d14bf73b4348c6e0d458b50d0109c85/tests/test_postgresql_provider.py#L248